org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/servicetunnel/HttpServiceTunnel.java

```diff
@@ -192,21 +192,15 @@ protected ServiceTunnelResponse tunnel(final ServiceTunnelRequest serviceRequest
       final byte[] requestData = requestMessage.toByteArray();
       nBytes = requestData.length;
 
-      // Send the request to the server.
-      try (Response resp = executeRequest(serviceRequest, requestData)) {
-        interceptHttpResponse(resp, serviceRequest);
-
-        try (InputStream in = resp.readEntity(InputStream.class)) {
-          // Receive the response.
-          ServiceTunnelResponse response = contentHandler.readResponse(in);
-          if (response == null) {
-            return new ServiceTunnelResponse(new ProcessingException("Response contains no content")
-                .withContextInfo("http-status", "{} {}", resp.getStatus(), resp.getStatusInfo())
-                .withContextInfo("http-headers", resp.getHeaders() + ""));
-          }
-          return response;
-        }
+      Response response = executeRequest(serviceRequest, requestData);  // response will be closed by #readResponse
+      try {
+        interceptHttpResponse(response, serviceRequest);
+      }
+      catch (Throwable t) {
+        response.close(); // close response in case of exception in #interceptHttpResponse
+        throw t;
       }
+      return readResponse(response, contentHandler);
```

Comment:
```suggestion
        return readResponse(response, contentHandler);
      }
      catch (Throwable t) {
        response.close(); // close response in case of exception in #interceptHttpResponse
        throw t;
      }
```

Not sure if `readResponse` should be within `try` block (what if an exception occurs during `readResponse`, do we not have to close it as well)? At least our implementation of `readResponse` seems to handle this case and has a try-with-resources block around the `response` - maybe not every implementation of `readResponse` will have this `try-with-resources` block for the response (as it will always close the response) but at least in case of an error is it expected that the response will be closed always? Maybe we could also add it to javadoc of `readResponse`?

Reply:
Moved readResponse within try/catch in order to close response in case of an error.